nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ParseCEF.java

```diff
@@ -0,0 +1,323 @@
+    @WritesAttribute(attribute = "cef.header.deviceVendor", description = "The Device Vendor of the CEF message."),
+    @WritesAttribute(attribute = "cef.header.deviceProduct", description = "The Device Product of the CEF message."),
+    @WritesAttribute(attribute = "cef.header.deviceVersion", description = "The Device Version of the CEF message."),
+    @WritesAttribute(attribute = "cef.header.deviceEventClassId", description = "The Device Event Class ID of the CEF message."),
+    @WritesAttribute(attribute = "cef.header.name", description = "The name of the CEF message."),
+    @WritesAttribute(attribute = "cef.header.severity", description = "The severity of the CEF message."),
+    @WritesAttribute(attribute = "cef.extension.*", description = "The key and value generated by the parsing of the message.")})
+@SeeAlso({ParseSyslog.class})
+
+public class ParseCEF extends AbstractProcessor {
+
+    // There should be no date format other than internationally agreed formats...
+    // flowfile-attributes uses Java 8 time to parse data (as Date  objects are not timezoned)
+    private final static DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
+
+    // for some reason Jackson doesnt seem to be able to use DateTieFormater
+    // so we use a SimpleDateFormat to format within flowfile-content
+    private final SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
+
+
+    // add a TZ object to be used by flowfile-attribute routine
+    private String tzId = null;
+
+    // Add serializer and mapper
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    public static final String DESTINATION_CONTENT = "flowfile-content";
+    public static final String DESTINATION_ATTRIBUTES = "flowfile-attribute";
+    public static final PropertyDescriptor FIELDS_DESTINATION = new PropertyDescriptor.Builder()
+        .name("FIELDS_DESTINATION")
+        .displayName("Parsed fields destination")
+        .description(
+                "Indicates whether the results of the CEF parser are written " +
+                "to the FlowFile content or a FlowFile attribute; if using " + DESTINATION_ATTRIBUTES +
+                "attribute, fields will be populated as attributes. " +
+                "If set to " + DESTINATION_CONTENT + ", the CEF extension field will be converted into " +
+                "a flat JSON object.")
+        .required(true)
+        .allowableValues(DESTINATION_CONTENT, DESTINATION_ATTRIBUTES)
+        .defaultValue(DESTINATION_CONTENT)
+        .build();
+
+    public static final PropertyDescriptor APPEND_RAW_MESSAGE_TO_JSON = new PropertyDescriptor.Builder()
+            .name("APPEND_RAW_MESSAGE_TO_JSON")
+            .displayName("Append raw message to JSON")
+            .description("When using flowfile-content, add the original flow content as JSON node _raw, " +
```

Comment:
I reworded the description a bit to clarify the intent of the setting.

The characters should be escaped by jackson when added as an element to the object. 

Added a unit test to cover that scenario.

Failed flowfiles as passed through with no modification to content.